internal/pkg/utils/time.go

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright (C) 2023 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/errors"
+)
+
+// CheckDuration parses the ISO 8601 time duration string to Duration type
+// and evaluates if the duration value is smaller than the suggested minimum duration string
+func CheckDuration(value string, min string) (bool, errors.EdgeX) {
```

Comment:
rename to `CheckMinInterval`, because it's not general for all kinds of Duration check, and it doesn't need to return error.
Keep the bool return value for unit test, but the caller can ignore it.